docs/section-3-creating-parsers.md

```diff
@@ -557,6 +557,8 @@ Although languages have very different constructs, their constructs can often be
 }
 ```
 
+One important thing to note up front is that the entry point for grammar parsing is defined as the first property in the `rules` object. The formal specification for what constitutes the "first property" is found in the ECMAScript 2015 specification. However, since grammars typically only contain string-based property names, it is safe to assume that the top-most key-value pair in `rules` will be treated as the entry point. In the examples on this page, this rule is called `source_file`, but it can be anything. [Other](https://github.com/tree-sitter/tree-sitter-haskell/blob/2c2fcb22e3b939ed59ae8d6f10500c3213c68670/grammar.js) [grammars](https://github.com/kylegoetz/tree-sitter-unison/blob/3c97db76d3cdbd002dfba493620c2d5df2fd6fa9/grammar.js) have named this property after the programming language whose grammar is described by the file.
```

Comment:
Too verbose.
```suggestion
One important thing to note up front is that the entry point for grammar parsing is defined as the first property in the `rules` object. In the examples on this page, this rule is called `source_file`, but it can be anything.
```